Plots/Panels/NCL_dev_1.py

```diff
@@ -0,0 +1,153 @@
+"""
+
+##############################################################################
+# Import packages:
+import cartopy.crs as ccrs
+from cartopy.mpl.gridliner import LongitudeFormatter, LatitudeFormatter
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+
+import geocat.datafiles as gdf
+from geocat.viz import cmaps as gvcmaps
+import geocat.viz.util as gvutil
+
+##############################################################################
+# Read in data:
+
+# Open a netCDF data file using xarray default engine and load the data into xarrays
+ds = xr.open_dataset(gdf.get("netcdf_files/83.nc"))
+# Extract slice of data
+TS = ds.TS.isel(time=0).drop('time')
+# Fix the artifact of not-shown-data around 0 and 360-degree longitudes
+TS = gvutil.xr_add_cyclic_longitudes(TS, "lon")
+
+# Calculate zonal mean
+mean = TS.mean(dim='lon')
+
+# Using meshgrid, a 2-D array can be created with the same shape as the
+# temperature data with the zonal mean for each latitude filling each row.
+# This way we can subtract each element of the mean 2-D array from the
+# corresponding element in the data array.
+waste, mean_grid = np.meshgrid(TS['lon'], mean)
+# Calculate deviations from zonal mean
+dev = TS.data - mean_grid
+
+##############################################################################
+# Plot:
+
+# Specify projection for maps
+proj = ccrs.PlateCarree()
+
+# Generate figure (set its size (width, height) in inches)
+fig = plt.figure(figsize=(8, 8))
+grid = fig.add_gridspec(ncols=2, nrows=2, width_ratios=[0.85, 0.15],
+                        wspace=0.08)
+
+# Create axis for original data plot
+ax1 = fig.add_subplot(grid[0, 0], projection=ccrs.PlateCarree())
+ax1.coastlines(linewidths=0.5)
+
+# Create axis for zonal mean plot
+ax2 = fig.add_subplot(grid[0, 1], aspect=5.9)
+
+# Create axis for deviation data plot
+ax3 = fig.add_subplot(grid[1, 0], projection=ccrs.PlateCarree())
+ax3.coastlines(linewidths=0.5)
+
+# Create axis for colorbar
+ax4 = fig.add_subplot(grid[1, 1], aspect=10)
+
+# Format ticks and ticklabels for the map axes
+for ax in [ax1, ax3]:
+    # Use the geocat.viz function to set axes limits and ticks
+    gvutil.set_axes_limits_and_ticks(ax, xlim=[-180, 180], ylim=[-90, 90],
+                                     xticks=np.arange(-180, 181, 30),
+                                     yticks=np.arange(-90, 91, 30))
+    
+    # Use the geocat.viz function to add minor ticks
+    gvutil.add_major_minor_ticks(ax)
+    
+    # Use geocat.viz.util convenience function to make plots look like NCL
+    # plots by using latitude, longitude tick labels
+    gvutil.add_lat_lon_ticklabels(ax)
+    
+    # Removing degree symbol from tick labels to more closely resemble NCL example
+    ax.yaxis.set_major_formatter(LatitudeFormatter(degree_symbol=''))
+    ax.xaxis.set_major_formatter(LongitudeFormatter(degree_symbol=''))
+
+
+# Format ticks and ticklabels for zonal average plot
```

Comment:
Is the "Format" comment tied to the "Use" comment?

Reply:
I can combine the two comments for clarity.